gcloud/storage/bucket.py

```diff
@@ -854,18 +869,26 @@ def make_public(self, recursive=False, future=False):
         :type future: boolean
         :param future: If True, this will make all objects created in the
                        future public as well.
+
+        :type connection: :class:`gcloud.storage.connection.Connection` or
+                          ``NoneType``
+        :param connection: Optional. The connection to use when sending
+                           requests. If not provided, falls back to default.
         """
+        connection = _require_connection(connection)
+
         self.acl.all().grant_read()
-        self.acl.save()
+        self.acl.save(connection=connection)
 
         if future:
             doa = self.default_object_acl
             if not doa.loaded:
-                doa.reload()
+                doa.reload(connection=connection)
             doa.all().grant_read()
-            doa.save()
+            doa.save(connection=connection)
 
         if recursive:
-            for blob in self:
+            for blob in self.list_blobs(projection='full',
+                                        connection=connection):
                 blob.acl.all().grant_read()
-                blob.save_acl()
+                blob.acl.save(connection=connection)
```

Comment:
I don't think `make_public` is insane for its intended usecase, which is to build out a static website and then publish it all at once.  Such a site isn't going to have millions of blobs.